sos/report/plugins/__init__.py

```diff
@@ -2954,8 +2954,8 @@ def filter_namespaces(self, ns_list, ns_pattern=None, ns_max=None):
         for ns in ns_list:
             # if ns_pattern defined, skip namespaces not matching the pattern
             if ns_pattern:
-                if not bool(re.match(pattern, ns)):
-                    continue
+                if bool(re.match(pattern, ns)):
+                    out_ns.append(ns)
```

Comment:
I don't think this is right. We're issuing a `continue` here because we aren't matching the pattern. We don't directly add the namespace to `out_ns` here because there are still other criteria.

I think instead we need to change separate this conditional, rather than a continuation of the pattern check
 https://github.com/sosreport/sos/blob/dd8defdeec62a6f086bc2afebfb87b4af74178c0/sos/report/plugins/__init__.py#L2962 - i.e. make `elif` a new `if` statement.

Reply:
Right; my patch works with the (wrong) assumption that the pattern and max can not be set concurrently (at least the original code suggested me that assumption). So when the options are used in conjunction, `ns_max` is ignored.

Updating the PR accordingly.